lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -228,6 +252,25 @@ def message_from(action, response)
         response['status_detail'] || response['message']
       end
 
+      def success_from_get(response)
+        if response.instance_of? Array
+          return true
+        end
```

Comment:
This will only be true for `/payment-methods` API. Can we check for status_code alone in the response?

Reply:
Addressed.